tests/unit_tests/test_cell.py

```diff
@@ -112,14 +123,118 @@ def test_get_nuclides(uo2):
     assert nucs == ['U235', 'O16']
 
 
+def test_volume_setting():
+    c = openmc.Cell()
+
+    # Test ordinary volume and uncertain volume
+    c.volume = 3
+    c.volume = u.ufloat(3, 0.7)
+
+    # Test errors for -ve and 0 volume
+    with pytest.raises(ValueError):
+        c.volume = 0.0
```

Comment:
While it isn't useful and probably never happens intentionally, it is possible to create a cell with zero volume.

Additionally, I can see this case causing problems when results from a volume calculation are applied and no samples happened to fall in that cell - either b/c it is very small or there is some error in the geometry.

Reply:
Hrmm, so it [looks like](https://github.com/openmc-dev/openmc/blob/8154e5a0350bf6ea342688f8b721cad84b60dfdb/openmc/cell.py#L302) we set the volume of a cell directly when applying results from a volume calculation. So this won't throw an exception when applying volume results, but it might appear inconsistent to users for some cells to have zero volume after a volume calculation while it is impossible to set that value manually.

Reply:
You are absolutely right. I have completely missed the possibility that cell can have a 0.0 volume as a result of stochastic volume calculation. I have changed the `volume` setter and unit test to accept the volume==0.0.